File: python/algorithms/easy/birthdayCakeCandles.py
"""
Author: Rithie Natan Carvalhaes Prado
Date: 2026-06-15
Last modified: 2026-06-15
Description: Birthday Cake Candles - HackerRank
Test description: https://www.hackerrank.com/challenges/birthday-cake-candles/problem
"""
#---------- functions ----------
def birthdayCakeCandles(candles: list[int]) -> int:
    """
    Count how many candles are tallest in the array of candles. The tallest candles are the ones that can be blown out.

    Parameters:
        candles (list[int]): List of candle heights.

    Returns:
        int: Count of the tallest candles.
    """
    count = 1
    candles.sort(reverse=True)
    lastNumber = candles[0]

    for i in range(1, len(candles)):
        if candles[i] == lastNumber:
            count += 1
        else:
            break

    return count

File: go/algorithms/easy/birthdayCakeCandles.go
/*
Author: Rithie Natan Carvalhaes Prado
Date: 2026-06-15
Last modified: 2026-06-15
Description: Birthday Cake Candles - HackerRank
See: https://www.hackerrank.com/challenges/birthday-cake-candles/problem
*/
//--------- package ---------
package easy


//--------- imports ---------
import (
	"cmp"
	"slices"
)


//--------- functions ---------
// Count the number of candles that are tallest.
func birthdayCakeCandles(candles []int32) int32 {
	var count int32 = 1;
	slices.SortFunc(candles, func(a, b int32) int {
		return cmp.Compare(b, a);
	});
	lastNumber := candles[0];

	for i := 1; i < len(candles); i++ {
        if lastNumber == candles[i] {
            count++;
        } else {
            break;
        }
    }//end for
    
    return count;
}//end birthdayCakeCandles()